File: backend/src/main/java/com/skewwhiffy/auraltester/notation/model/scale/ScaleType.java
package com.skewwhiffy.auraltester.notation.model.scale;

import com.skewwhiffy.auraltester.notation.model.interval.DirectedInterval;

import java.util.List;

public record ScaleType(
        String displayName,
        String abc,
        List<DirectedInterval> ascendingIntervals,
        List<DirectedInterval> descendingIntervals
) {
    public ScaleType(String displayName, String abc, List<DirectedInterval> intervals) {
        this(displayName, abc, intervals, intervals);
    }

    public List<DirectedInterval> getIntervals(ScaleDirection direction) {
        return switch (direction) {
            case ASCENDING -> ascendingIntervals;
            case DESCENDING -> descendingIntervals;
        };
    }
}

File: backend/src/main/kotlin/com/skewwhiffy/auraltester/notation/model/scale/ScaleType.kt
package com.skewwhiffy.auraltester.notation.model.scale

import com.skewwhiffy.auraltester.notation.model.interval.DirectedInterval


data class ScaleType(
    val displayName: String,
    val abc: String,
    val ascendingIntervals: List<DirectedInterval>,
    val descendingIntervals: List<DirectedInterval>
) {
    constructor(displayName: String, abc: String, intervals: List<DirectedInterval>) : this(
        displayName,
        abc,
        intervals,
        intervals
    )

    fun getIntervals(direction: ScaleDirection) = when (direction) {
        ScaleDirection.ASCENDING -> ascendingIntervals
        ScaleDirection.DESCENDING -> descendingIntervals
    }
}
